### PR TITLE
Fix: Add training set size to log file

### LINKED ISSUE #583: Add training set size to log file
The log file should indicate the total number of spectra in the training set.  I'm not sure if this information is available at the start of the run, so maybe it gets reported after the first epoch.

### PR DESCRIPTION
**Pull request by gabrnavarro**
# Add training set size to log file

## What changed

Log the number of training and validation spectra during model training setup.
After the datasets are created in `DeNovoDataModule.setup()`, the total number
of spectra for each dataset is logged via the casanovo logger.

## Why

Users need visibility into the size of their training and validation sets. This
information is useful for monitoring and debugging training runs.

Fixes #583

## Testing

- Added `test_log_training_set_size` unit test that verifies training and
  validation spectra counts are logged when setting up the data module with
  `shuffle=False`.
- Added `test_log_training_set_size_shuffled` unit test that verifies the same
  logging works when `shuffle=True` (i.e. datasets wrapped in
  `ShufflerIterDataPipe`).
- All existing tests pass.

## Summary by CodeRabbit

* **Chores**
  * Enhanced logging during dataset initialization to display the count of training and validation spectra, including support for shuffled dataset configurations.

### CODE REVIEW

**Comment by gabrnavarro** [comment]
I've addressed the review feedback (attempt 2). Updated the branch with the requested changes.

**Comment by bittremieux** [comment]
@gabrnavarro Thank you for the contribution. Can you please do the following:
1. Rebase the PR against `dev` (instead of `main`).
2. Make sure that all unit tests complete successfully and that any potential CodeRabbit comments have been addressed.
3. After this is done, please tag me for a code review.

**Comment by bittremieux** [comment]
Stale PR, closing. Feel free to reopen the PR when you're ready to address the requested changes.
